flight/PiOS/Common/pios_px4flow.c

```diff
@@ -0,0 +1,286 @@
+/**
+ ******************************************************************************
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+ * @addtogroup PIOS_PX4FLOW PX4Flow Functions
+ * @brief Deals with the hardware interface to the PixHawk optical flow sensor
+ * @{
+ * @file       pios_px4flow.c
+ * @author     Kenn Sebesta, Copyright (C) 2014
+ * @brief      PX4Flow optical flow sensor
+ *
+ ******************************************************************************
+ */
+
+/* Project Includes */
+#include "pios.h"
+#include "coordinate_conversions.h"
+#include "physical_constants.h"
+
+#if defined(PIOS_INCLUDE_PX4FLOW)
+#include "pios_px4flow_priv.h"
+
+#include "pios_semaphore.h"
+#include "pios_thread.h"
+#include "pios_queue.h"
+
+/* Private constants */
+#define PX4FLOW_TASK_PRIORITY        PIOS_THREAD_PRIO_HIGH
+#define PX4FLOW_TASK_STACK_BYTES     512
+#define PX4FLOW_SAMPLE_PERIOD_MS     5
```

Comment:
The actual multitasking tick rate we have is 1KHz.. 400Hz isn't a factor of this, and turning it up too much further probably isn't good.  One could do 250Hz or 333Hz though...